libsolidity/analysis/ControlFlowBuilder.cpp

```diff
@@ -57,24 +57,61 @@ unique_ptr<FunctionFlow> ControlFlowBuilder::createFunctionFlow(
 bool ControlFlowBuilder::visit(BinaryOperation const& _operation)
 {
 	solAssert(!!m_currentNode, "");
+	visitNode(_operation);
 
 	switch (_operation.getOperator())
 	{
 		case Token::Or:
 		case Token::And:
 		{
 			visitNode(_operation);
```

Comment:
Wait - is this visited twice now? Is that intentional?

Reply:
Not that it matters, but a bit weird - the second one here can be removed, can't it?

Reply:
``ASTConstVisitor::visit(_operation);`` then does it yet another time...

Reply:
But well, it's not like it hurts to call it multiple times - and even if the source locations ended up incorrect, it'll still only result in weird error messages and not in actually broken behaviour, so fine either way.

Reply:
Need to take a closer look, but in the original PR was actually skipping some `visitNode()` calls that should have been there. Need to check if it's that or if not there's too many of them :)

Reply:
Looks like it was moved in and out of the switch in the original PR. There is even a comment about that (https://github.com/ethereum/solidity/pull/12362/files#r931895231). Something must have gone wrong with that because apparently it was added inside but not removed outside. I must have missed that in review too. There were actually a few changes like this that ended up marked as resolved but not applied and I had to ask about them.

In any case, I'm removing it.